login-frontend/src/App.tsx

```diff
@@ -55,13 +55,17 @@ export function App() {
           }
         });
 
-        let url = '/api/v1/credentials';
         // get redirect uri query
-        const redirectUri = new URLSearchParams(window.location.search).get('redirect');
-        if (redirectUri) {
-          url = `${url}?redirect=${redirectUri}`;
+        if (!fetchedUri) {
+          const redirectUri = new URLSearchParams(window.location.search).get('redirect');
+          if (redirectUri) {
+            setFetchedUri(`/api/v1/credentials?redirect=${redirectUri}`);
+          } else {
+            console.error('Redirect URI is not present, please go back to original URL');
+            setError(true);
+          }
```

Comment:
I don't think adding this new state variable is useful. You're still just checking if the redirect parameter is present in the URL. Your state variable will be reset when the page is refreshed.